c7n/resources/guardduty.py

```diff
@@ -0,0 +1,40 @@
+# Copyright The Cloud Custodian Authors.
+# SPDX-License-Identifier: Apache-2.0
+from c7n.manager import resources
+from c7n import query
+from c7n.utils import local_session
+
+
+class DescribeGuarddutyFinding(query.DescribeSource):
+
+    def resources(self, query):
+        detector_id = self.get_detector_id()
+        if not detector_id:
+           return ()
+        if not query:
+           query = {}
+        query['DetectorId'] = detector_id
+        return super().resources(query)
+
+    def get_detector_id(self):
+        client = local_session(self.manager.session_factory).client('guardduty')
+        ids = client.list_detector_ids().get('DetectorIds')
+        return ids and ids[0] or ids
+
+
+@resources.register("aws.guardduty-finding")
+class GuarddutyFinding(query.QueryResourceManager):
+    class resource_type(query.TypeInfo):
+        service = "guardduty"
+        enum_spec = ('list_findings', 'FindingIds', None)
+        detail_spec = ('get_findings', 'FindingIds', None, 'Findings')
```

Comment:
This dies twice for me, once because `DetectorId` is required and again because `FindingIds` needs to be a list:

```
botocore.exceptions.ParamValidationError: Parameter validation failed:
Missing required parameter in input: "DetectorId"
Invalid type for parameter FindingIds, value: ..., type: <class 'str'>, valid types: <class 'list'>, <class 'tuple'>
```

Reply:
its using a custom source re detector id

Reply:
Yeah, it looked like the custom source got the enum sorted out but that `DetectorId` kwarg didn't carry through to the scalar augment call [here](https://github.com/cloud-custodian/cloud-custodian/blob/34dd36a45e114c996941bb81901c9db6013ea55e/c7n/query.py#L747).

I didn't include the full traceback because it was pretty noisy, but here goes...

<summary>full traceback</summary>

<detail>

```
2023-08-07 19:13:11,385: custodian.commands:ERROR Error while executing policy aws-guard-duty-findings, continuing
Traceback (most recent call last):
  File "/home/aj/code/cloud-custodian/c7n/commands.py", line 307, in run
    policy()
  File "/home/aj/code/cloud-custodian/c7n/policy.py", line 1352, in __call__
    resources = PullMode(self).run()
                ^^^^^^^^^^^^^^^^^^^^
  File "/home/aj/code/cloud-custodian/c7n/policy.py", line 330, in run
    resources = self.policy.resource_manager.resources()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/aj/code/cloud-custodian/c7n/query.py", line 543, in resources
    resources = self.augment(resources)
                ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/aj/code/cloud-custodian/c7n/query.py", line 599, in augment
    return self.source.augment(resources)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/aj/code/cloud-custodian/c7n/query.py", line 275, in augment
    results = list(w.map(
              ^^^^^^^^^^^
  File "/home/aj/.pyenv/versions/3.11.0/lib/python3.11/concurrent/futures/_base.py", line 619, in result_iterator
    yield _result_or_cancel(fs.pop())
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/aj/.pyenv/versions/3.11.0/lib/python3.11/concurrent/futures/_base.py", line 317, in _result_or_cancel
    return fut.result(timeout)
           ^^^^^^^^^^^^^^^^^^^
  File "/home/aj/.pyenv/versions/3.11.0/lib/python3.11/concurrent/futures/_base.py", line 456, in result
    return self.__get_result()
           ^^^^^^^^^^^^^^^^^^^
  File "/home/aj/.pyenv/versions/3.11.0/lib/python3.11/concurrent/futures/_base.py", line 401, in __get_result
    raise self._exception
  File "/home/aj/.pyenv/versions/3.11.0/lib/python3.11/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/aj/code/cloud-custodian/c7n/query.py", line 747, in _scalar_augment
    response = op(*args, **kw)
               ^^^^^^^^^^^^^^^
  File "/home/aj/code/cloud-custodian/c7n/utils.py", line 452, in _retry
    return func(*args, **kw)
           ^^^^^^^^^^^^^^^^^
  File "/home/aj/.pyenv/versions/3.11.0/envs/c7n-dev/lib/python3.11/site-packages/botocore/client.py", line 534, in _api_call
    return self._make_api_call(operation_name, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/aj/.pyenv/versions/3.11.0/envs/c7n-dev/lib/python3.11/site-packages/botocore/client.py", line 935, in _make_api_call
    request_dict = self._convert_to_request_dict(
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/aj/.pyenv/versions/3.11.0/envs/c7n-dev/lib/python3.11/site-packages/botocore/client.py", line 1003, in _convert_to_request_dict
    request_dict = self._serializer.serialize_to_request(
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/aj/.pyenv/versions/3.11.0/envs/c7n-dev/lib/python3.11/site-packages/botocore/validate.py", line 381, in serialize_to_request
    raise ParamValidationError(report=report.generate_report())
botocore.exceptions.ParamValidationError: Parameter validation failed:
Missing required parameter in input: "DetectorId"
Invalid type for parameter FindingIds, value: 08c4cbd0556ee22135d8be755e56dee6, type: <class 'str'>, valid types: <class 'list'>, <class 'tuple'>
2023-08-07 19:13:11,388: custodian.commands:ERROR The following policies had errors while executing
 - aws-guard-duty-findings
 
 ```
 
 </detail>

Reply:
thanks I'll investigate further.

Reply:
fixed